flocker/ca/functional/test_validation.py

```diff
@@ -74,6 +81,22 @@ def getContext(self):
         return ctx
 
 
+class WaitForDisconnectsFactory(ServerFactory):
+    """
+    A factory for use with ``SendingProtocol`` that makes it possible to wait
```

Comment:
Does this set us up for timeouts if something does not disconnect? If so I guess we won't have clear error messages telling us what went wrong.